src/modules/75-cd/pages/get-started-with-cd/CDOnboardingWizardWithCLI/Steps/WhereAndHowToDeploy.tsx

```diff
@@ -99,21 +100,38 @@ function WhereAndHowToDeploy({ saveProgress }: WhereAndHowToDeployProps): JSX.El
   const onDelegateSuccess = (): void => {
     onChangeHandler('SUCCESS')
     trackEvent(ONBOARDING_INTERACTIONS.DELEGATE_VERIFICATION_SUCCESS, {
-      ...getBranchingProps(stepsProgress),
+      ...getBranchingProps(stepsProgress, getString),
       delegateName: state?.delegateName,
       delegateType: state?.delegateType
     })
   }
 
   const onVerificationStart = (): void => {
+    onChangeHandler('TRYING')
     trackEvent(ONBOARDING_INTERACTIONS.DELEGATE_VERIFICATION_START, {
-      ...getBranchingProps(stepsProgress),
+      ...getBranchingProps(stepsProgress, getString),
       delegateName: state?.delegateName,
       delegateType: state?.delegateType
     })
   }
   const setType = (selectedType: DeploymentFlowType): void => {
-    setState(prevState => ({ ...prevState, type: selectedType }))
+    const clearGitops =
+      selectedType?.id === DEPLOYMENT_FLOW_ENUMS.CDPipeline && state.type?.id !== DEPLOYMENT_FLOW_ENUMS.CDPipeline
+    if (clearGitops) {
+      setState(prevState => ({ ...prevState, type: selectedType, agentInfo: undefined }))
+    } else {
+      setState(prevState => ({ ...prevState, type: selectedType }))
+    }
+  }
+  const onAgentVerificationSuccess = (status?: Servicev1HealthStatus): void => {
+    setState((prevState: WhereAndHowToDeployType) => ({ ...prevState, agentStatus: status }))
+  }
+  const updateAgentInfo = (agentDetails: V1Agent): void => {
+    setState((prevState: WhereAndHowToDeployType) => ({
+      ...prevState,
+      agentInfo: agentDetails,
+      installAgentTried: agentDetails ? true : false
```

Comment:
You can simplify a bit
```suggestion
      installAgentTried: !!agentDetails
```